src/gateway/router/routes/sequencerRoute.ts

```diff
@@ -57,22 +52,43 @@ export async function sequencerRoute(ctx: Router.RouterContext) {
       throw new Error('Current block not set');
     }
 
-    const millis = Date.now();
-    const sortKey = await createSortKey(arweave, jwk, currentBlockId, millis, transaction.id, currentHeight);
-    let { contractTag, inputTag, internalWrites, decodedTags, tags, vrfData, originalAddress, isEvmSigner, testnetVersion } = await prepareTags(
+    let {
+      contractTag,
+      inputTag,
+      requestVrfTag,
+      internalWrites,
+      decodedTags,
+      tags,
+      originalAddress,
+      isEvmSigner,
+      testnetVersion
+    } = await prepareTags(
       sLogger,
       transaction,
       originalOwner,
-      millis,
-      sortKey,
       currentHeight,
       currentBlockId,
-      vrf,
       arweave
     );
 
+    contractMutexRelease = await lastTxSync.acquireMutex(contractTag);
+    const contractLastSortKey: string  | null = await lastTxSync.getLastSortKey(contractTag);
```

Comment:
What if we cannot get lastSortKey because of the mutex'es issues?
We don't handle any error there - should we retry it somehow?

Reply:
in case of any issues with either acquiring a mutex or load lastSortKey - an error will be thrown and catched in https://github.com/warp-contracts/gateway/pull/110/files#diff-925002587e7eba50a7804e4021c11fc908454e00be475613a591a693fdb4723fR165

I believe 'fail-fast' is the best approach, no retries etc. for simplicy - we can add them in Go version.